modules/ggl-uri/src/gghttp_uri.c

```diff
@@ -136,3 +138,283 @@ GglError gg_uri_parse(GglArena *arena, GglBuffer uri, GglUriInfo *info) {
 
     return GGL_ERR_OK;
 }
+
+static GglError find_docker_uri_separators(
+    GglBuffer uri,
+    size_t slashes[2],
+    size_t *slash_count,
+    size_t colons[3],
+    size_t *colon_count,
+    size_t *at,
+    bool *has_registry
+) {
+    for (size_t i = uri.len; i > 0; i--) {
+        if (uri.data[i - 1] == '/') {
+            if (*slash_count < 2) {
+                slashes[*slash_count] = i - 1;
+                *slash_count += 1;
+                GGL_LOGT("Found a slash while parsing Docker URI");
+                continue;
+            }
+            GGL_LOGE("More than two slashes found while parsing Docker URI, "
+                     "URI is invalid.");
+            return GGL_ERR_INVALID;
+        }
+        if (uri.data[i - 1] == ':') {
+            if (*colon_count < 3) {
+                colons[*colon_count] = i - 1;
+                *colon_count += 1;
+                GGL_LOGT("Found a colon while parsing Docker URI");
+                continue;
+            }
+            GGL_LOGE("More than three colons found while parsing Docker URI, "
+                     "URI is invalid.");
+            return GGL_ERR_INVALID;
+        }
+        if (uri.data[i - 1] == '@') {
+            if (*at == SIZE_MAX) {
```

Comment:
I'd much rather maintain a local count instead of the caller setting the correct value of `at` variable to `SIZE_MAX`.